pkg/ccl/kvccl/kvtenantccl/upgradeccl/tenant_upgrade_test.go

```diff
@@ -195,15 +194,25 @@ func v0v1v2() (roachpb.Version, roachpb.Version, roachpb.Version) {
 // between version upgrades.
 func TestTenantUpgradeFailure(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	skip.WithIssue(t, 98555, "flaky test")
 	defer log.Scope(t).Close(t)
 	// Contains information for starting a tenant
 	// and maintaining a stopper.
 	type tenantInfo struct {
 		v2onMigrationStopper *stop.Stopper
 		tenantArgs           *base.TestTenantArgs
 	}
-	v0, v1, v2 := v0v1v2()
+	v0 := clusterversion.TestingBinaryMinSupportedVersion
+	v2 := clusterversion.TestingBinaryVersion
+	// v1 needs to be between v0 and v2. Set it to the minor release
+	// after v0 and before v2.
+	var v1 roachpb.Version
+	for _, version := range clusterversion.ListBetween(v0, v2) {
+		if version.Minor != v0.Minor {
+			v1 = version
+			break
+		}
+	}
```

Comment:
Can you add an assert that v1 was assigned, in case the loop above turned out empty.